doc/leak_tracking/DETECT.md

```diff
@@ -0,0 +1,165 @@
+2. Before `runApp` invocation, enable leak tracking, and connect
+the Flutter memory allocation events:
+
+```dart
+import 'package:flutter/foundation.dart';
+import 'package:leak_tracker/leak_tracker.dart';
+
+...
+
+enableLeakTracking();
+FlutterMemoryAllocations.instance
+      .addListener((ObjectEvent event) => dispatchObjectEvent(event.toMap()));
+runApp(...);
+
+```
+
+3. Run the application in debug mode and watch for a leak related warnings.
+
+TODO(polina-c): add example of the warning.
+
+## Limitations
+
+### By environment
+
+At this time, leak tracking is supported only for unit tests within Flutter packages.
+
+With the latest version of leak tracker, leak tracking is not supported for:
+
+1. Web platform
+2. Running applications
+3. Pure Dart packages
+
+### By leak types
+
+Leak tracking for not-GCed leaks is experimental and is off by default.
+At this time, it is recommended to track only not-disposed leaks.
+
+### By tracked classes
+
+The leak tracker will catch leaks only for instrumented
+objects (see [concepts](CONCEPTS.md) for details).
+
+However, the good news is:
+
+1. Disposables in the Flutter Framework are instrumented.
+If your Flutter app manages widgets in a way that results in leaks,
+`leak_tracker` will catch them.
+
+2. If a leak involves at least one instrumented object,
+the leak will be caught and all
+other objects, even non-instrumented, will stop leaking as well.
+
+See [the instrumentation guidance](#instrument-your-code).
+
+### By build mode
+
+The leak tracker availability differs by build modes.
+See [Dart build modes](https://github.com/dart-lang/site-www/issues/4436)
+or [Flutter build modes](https://docs.flutter.dev/testing/build-modes).
+
+**Dart development and Flutter debug**
+
+Leak tracking is fully available.
+
+**Flutter profile**
+
+Leak tracking is available, but `FlutterMemoryAllocations` that listens to
+Flutter instrumented objects,
+should be [turned on](https://github.com/flutter/flutter/blob/15af81782e19ebe7273872f8b07ac71df4e749f2/packages/flutter/lib/src/foundation/memory_allocations.dart#L13)
+if you want to track Flutter Framework objects.
+
+**Dart production and Flutter release**
+
+Leak tracking is disabled.
+
+NOTE: If you are interested in enabling leak tracking for release mode, please, comment [here](https://github.com/dart-lang/leak_tracker/issues/25).
+
+## Performance impact
+
+### Memory
+
+The Leak Tracker stores a small additional record for each
+tracked alive object and for each
+detected leak, that increases the memory footprint.
+
+For the [Gallery application](https://github.com/flutter/gallery)
+in profile mode on `macos`
+the leak tracking increased memory footprint of the home page
+by ~400 KB that is ~0.5% of
+the total.
+
+### CPU
+
+Leak tracking impacts CPU in two areas:
+
+1. Per object tracking.
+   Added ~0.05 of millisecond (~2.7%) to the total load time of
+   [Gallery](https://github.com/flutter/gallery) home page
+   in profile mode on `macos`.
```

Comment:
It would be more illustrative to measure the impact on profile mode on a mid-tier Android device.

Desktop machines tend to be significantly more powerful than mobile, and iPhone's tend to have better CPU/memory performance than Android.

Reply:
Yes. But while leak tracking is test-only, it is ok not to have these numbers.